src/server.ts

```diff
@@ -22,6 +40,18 @@ router.get('/test', (req: any, res: any) => res.send("Hello World!"));
 
 router.get("/rovers", (req, res) => getRovers(res));
 
+router.get("/rovers/:rovername/photos/:cameratype", (req, res: any) => {
+    let rovername: rover = rover[req.params["rovername"] as keyof typeof rover] // makes sure rover name is in list of rovers
+    let cameratype: camera = camera[req.params["cameratype"] as keyof typeof camera] // makes sure camera type is in list of camera types

```

Comment:
Using `as` with Typescript is you "overwriting" typescript and saying that you're expecting whatever to be the same `as` another typ.. Generally you don't want to be doing this unless you're certain it will be that type. In this case you're kind of using it as a validation thing (and at runtime you'll get a failure if a user doesn't give a valid rover name)

Instead you want to be more explicit about validation. For some backend frameworks, you can have the validation built into the endpoint (I think express has a validation package too). In our case you can do something like:
- check the rover param is the correct type, if not return a 400 bad request or 404 not found (worth checking which would be best here)
- check the camera type and do the same
- if no failures then then do getPhotos()

This way you're doing explicit validation rather than relying on `keyof typeof` shenanigans

Reply:
@SChandarana So I've tried 

`if (req.params["rovername"] in Rover && req.params["cameratype"] in Camera) {
      const keyRover: keyof typeof Rover = req.params["rovername"];
      const rovername: Rover = Rover[keyRover];

      const keyCamera: keyof typeof Camera = req.params["cameratype"];
      const cameratype: Camera = Camera[keyCamera];

      getPhotos(rovername, cameratype, res);
`
with and without the `keyof typeof` but it doesn't like that I'm trying to use a string to get the keys of my enum regardless (the string is not assignable to the given keys I have for that enum). The things I've looked at online use `as` to get around this. I guess I don't need the rover name to be of type `Rover` after I have checked that it is one of the possible rover names, so should I not try to cast to type Rover at all or would that be bad practice?